### PR TITLE
CO-2789: Fix multiple creation of "New Dossier" communication

### PR DESCRIPTION
**Pull request by Nooka10**

### CODE REVIEW

**Comment by ecino** [comment]
Please rebase on compassion/devel and remove unrelated commits. Fix lint issues and tests that are failing:
```
   File "/home/travis/build/CompassionCH/compassion-switzerland/partner_communication_switzerland/models/contracts.py", line 492, in contract_waiting
`     lambda c: 'S' in c.type and not c.is_active and c not in
`   File "/home/travis/build/CompassionCH/compassion-switzerland/partner_communication_switzerland/models/contracts.py", line 588, in _new_dossier
`     spo.send_communication(new_dossier)
`   File "/home/travis/build/CompassionCH/compassion-switzerland/partner_communication_switzerland/models/contracts.py", line 192, in send_communication
`     same_communication_search)
`   File "/home/travis/odoo-10.0/odoo/models.py", line 1516, in search
`     res = self._search(args, offset=offset, limit=limit, order=order, count=count)
`   File "/home/travis/odoo-10.0/odoo/models.py", line 4246, in _search
`     query = self._where_calc(args)
`   File "/home/travis/odoo-10.0/odoo/models.py", line 4040, in _where_calc
`     e = expression.expression(domain, self)
`   File "/home/travis/odoo-10.0/odoo/osv/expression.py", line 640, in __init__
`     self.expression = distribute_not(normalize_domain(domain))
`   File "/home/travis/odoo-10.0/odoo/osv/expression.py", line 289, in distribute_not
`     elif token in DOMAIN_OPERATORS_NEGATION:
` ValueError: <type 'exce
```

File: partner_communication_switzerland/models/contracts.py [hunk 1]
```diff
@@ -164,45 +164,34 @@ def send_communication(self, communication, correspondent=True,
...
-                    communications += self.env[
-                        'partner.communication.job'].create(
-                        {
-                            'config_id': communication.id,
-                            'partner_id': partner.id,
-                            'object_ids': self.env.context.get(
-                                'default_object_ids', objects.ids),
-                            'user_id': communication.user_id.id,
-                        }
-                    )
```
Comment (ecino) [inline partner_communication_switzerland/models/contracts.py]: You removed your fix!

File: partner_communication_switzerland/models/contracts.py [hunk 2]
```diff
@@ -445,14 +448,11 @@ def contract_waiting_mandate(self):
...
-                'sds_state': 'waiting_welcome', 'sds_state_date':
+                      s.partner_id.ref != '1502623' and not
+                      s.welcome_active_letter_sent).write({
+            'sds_state': 'waiting_welcome', 'sds_state_date':
                 fields.Date.today()})
-        csp = self.filtered(
-            lambda s: '6014' in s.mapped(
-                'contract_line_ids.product_id.property_account_income_id.code')
-        )
+        csp = self.filtered(lambda s: 'CSP' in s.name)
```
Comment (ecino) [inline partner_communication_switzerland/models/contracts.py]: This is a step back, it was on purpose changed to the above version. Please revert this change.

**Review by ecino — CHANGES_REQUESTED** [review]
Your last commit removed the purpose of this pull request :scream:

File: partner_communication_switzerland/models/contracts.py (hunk 1 above)
Comment (Nooka10) [reply partner_communication_switzerland/models/contracts.py]: Gosh... I need to sleep... -_-"

**Pull request closed without merging**